examples/game_of_life/measurement.py

```diff
@@ -0,0 +1,126 @@
+"""A default measurement with an array in and out."""
+import logging
+import pathlib
+import random
+import time
+
+import click
+
+import ni_measurementlink_service as nims
+from ni_measurementlink_service._internal.stubs.ni.protobuf.types import xydata_pb2
+
+service_directory = pathlib.Path(__file__).resolve().parent
+measurement_service = nims.MeasurementService(
+    service_config_path=service_directory / "game_of_life.serviceconfig",
+    version="1.0.0.0",
+    ui_file_paths=[service_directory / "game_of_life.measui"],
+)
+
+
+@measurement_service.register_measurement
+@measurement_service.configuration("width", nims.DataType.UInt32, 100)
+@measurement_service.configuration("height", nims.DataType.UInt32, 100)
+@measurement_service.configuration("update_interval", nims.DataType.UInt32, 100)
+@measurement_service.output("game_of_life", nims.DataType.DoubleXYData)
+@measurement_service.output("generation", nims.DataType.UInt32)
+def measure(width, height, update_interval):
+    """Streaming measurement that returns Conway's Game of Life grid as DoubleXYData."""
+    grid = _initialize_grid_with_seeded_data(width, height)
+    generation = 0
+    while True:
+        generation += 1
+
+        xydata = _initialize_xydata_and_frame(width, height)
+        row_index = 0
+        for row in grid:
+            col_index = 0
+            for cell in row:
+                if cell:
+                    xydata.x_data.append(row_index)
+                    xydata.y_data.append(col_index)
+                col_index += 1
+            row_index += 1
+        grid = _update_grid(grid)
+        xydata_out = xydata
+        generation += 1
+        time.sleep(update_interval / 1000)
+
+        yield (xydata_out, generation)
+
+
+def _initialize_xydata_and_frame(width, height) -> xydata_pb2.DoubleXYData:
+    xydata = xydata_pb2.DoubleXYData()
+
+    # Frame To keep the graph stable
+    xydata.x_data.append(-1)
+    xydata.y_data.append(-1)
+    xydata.x_data.append(-1)
+    xydata.y_data.append(height + 1)
+    xydata.x_data.append(width + 1)
+    xydata.y_data.append(-1)
+    xydata.x_data.append(width + 1)
+    xydata.y_data.append(height + 1)
+    return xydata
+
+
+def _initialize_grid_with_seeded_data(rows, cols, probability=0.6):
+    return [[random.random() < probability for _ in range(cols)] for _ in range(rows)]
+
+
+def _initialize_grid(rows, cols):
+    return [[False for _ in range(cols)] for _ in range(rows)]
```

Comment:
FYI, I was going to suggest `[[False] * cols] * rows` but unfortunately this is wrong because it returns N references to the same row.